src/checkpoint.hh

```diff
@@ -934,6 +978,117 @@ SUPPORTED_EXTERNAL_TYPES = {"Rope", "answer_pknot_mfe", "pktype",
      dec_indent(); dec_indent();
   }
 
+  void archive_cyk_indices(Printer::Base &stream, size_t n_tracks) {
+     inc_indent();
+     stream << indent() << "void archive_cyk_indices() {" << endl;
+     inc_indent();
+     stream << indent() << "// save the cyk loop indidces to disk" << endl;
+     stream << indent() << "try {" << endl;
+     inc_indent();
+     stream << indent() << "std::ofstream array_fout("
+            << "tmp_out_cyk_path.c_str()"
+            << ", std::ios::binary);" << endl;
+     stream << indent() << "if (!(array_fout.good())) {" << endl;
+     stream << indent() << "  throw std::ofstream::failure(\"\");" << endl;
+     stream << indent() << "}" << endl;
+     stream << indent() << "boost::archive::binary_oarchive "
+                            "array_out(array_fout);" << endl;
+     for (size_t i = 0; i < n_tracks; i++) {
+       // archive the indices of the prior iteration to ensure
+       // that the whole iteration has been completed
+       stream << indent() << "array_out << t_" << i << "_i + 1;" << endl;
+       stream << indent() << "array_out << "
+              << "(t_" << i << "_j == 0 ? 0 : t_" << i << "_j - 1);" << endl;
+     }
+     stream << indent() << "#ifdef _OPENMP" << endl;
+     stream << indent() << "  array_out << z1_;" << endl;
+     stream << indent() << "  array_out << z2_;" << endl;
+     stream << indent() << "  array_out << y_;" << endl;
```

Comment:
what are those z1_, z2_ and y variables? Did you introduce them or were they already present. Could we rename them into something more speaking?